core/controllers/domain_objects_validator.py

```diff
@@ -245,8 +245,9 @@ def validate_suggestion_images(files):
     return files
 
 
-def validate_exploration_change(exp_change_dict):
-    """Validate the exploration change list.
+def validate_exp_change_dict_and_return_exp_change_domain_obj(exp_change_dict):
```

Comment:
How about get_exp_change_from_dict() and have this function throw an error if the input is invalid?

Also the docstring says "change list" whereas the function name is "change dict", which is it?

Reply:
I have updated the name and the docstring. I already have a check for the invalid `cmd` part and the test regarding it. I think I will not be able to check for the other fields inside the change dict as the requirements for each of the field is different.
Thanks